src/dotnet/CarbonAware/ICarbonAware.cs

```diff
@@ -0,0 +1,21 @@
+using System.Collections;
+
+namespace CarbonAware;
+
+public interface ICarbonAware
```

Comment:
A thought occurs - is there a more precise name we could use for this interface? Like ```ICarbonAwarePlugin``` or even more like ```ICarbonAwareDataProvider```. I'm sorry I didn't think about this before in the design review but seeing it in usage made me think of this. Curious to hear your thoughts here

Reply:
@gfmatthews , this still in work in progress. We need to flush out more details here in terms of naming convention and other things. More to come :-)

Reply:
Sorry, this was not meant to be a public PR yet.. Still WIP as Juan said.